aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,1182 @@
+/// A work in progress NFT DAO Platform Example
+/// Note: Each token Id can only vote once as NFT is globally unique. If you create a DAO allowing fungible token to vote, only 1 of the fungible token can vote
+///
+/// An example e2e flow.
+/// There are multiple roles: DAO platform operator, DAO creator, proposer and voter.
+/// 1. platform operator deploy this package to create a DAO platform
+/// 2. DAO creator calls `create_dao` to create their DAO.
+/// 3. A proposer can specify the DAO she wants to create a proposal and create the proposal through  `create_proposal`
+///     two types of proposals can be created:
+///     3.a: no-op, no automatic execution happens on chain. Only the proposal and its results are recorded on-chain for DAO admin to take actions
+///     3.b: transfer fund, the fund will be automatically transferred from DAO resource account to the destination account.
+/// 4. A voter can vote for a proposal of a DAO through `vote`
+/// 5. Anyone can call the `resolve` to resolve a proposal. A proposal voting duration has to expire and the proposal should have more votes than the minimal required threshold.
+///
+/// An example of DAO delegation flow
+///  1. the DAO creator can delegate her DAO through offer_admin, claim_admin to another address
+///
+/// An example of DAO removal.
+/// 1. The DAO creator can call `reclaim_signer_capability` to remove her DAO from the platform and get back her resource account's signercapability
+///
+/// Note: This current version stores all the signer_capability in the contract space. If the code upgrades, a potential risk is the platform operation can obtain the signer capability of the DAOs on this platform
+/// It allows the operator to obtain the signer of these DAOs without any consent.
+/// TODO: explore new tooling to dynamically compiling DAO modules on server side and store the signercapability in each DAO's own moudle to avoid exposing signer capability to platform
+module dao_platform::nft_dao {
```

Comment:
it might be good to discuss the security semantics here... such as how do you ensure that a token cannot vote twice

Reply:
added in the comments and readme. record token_ids voted in bucket_table for each proposal and keyed on token_id